src/tribler/core/restapi/settings_endpoint.py

```diff
@@ -71,7 +71,13 @@ async def update_settings(self, request: web.Request) -> RESTResponse:
 
     def _recursive_merge_settings(self, existing: dict, updates: dict) -> None:
         for key in existing:
+            if key == "ui":
+                continue
```

Comment:
Could you double-check this? Looking at later code, I feel like you only need to ignore one "ui" key from the top level and not all "ui" keys in all nested dictionaries. I may be misinterpreting this though.

Reply:
Point taken, better to be safe then sorry. I've cooked up a fix for this.